packages/material-ui-lab/src/Breadcrumbs/Breadcrumbs.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { withStyles } from '@material-ui/core/styles';
+import { componentPropType } from '@material-ui/utils';
+import BreadcrumbCollapsed from './BreadcrumbCollapsed';
+import BreadcrumbSeparator from './BreadcrumbSeparator';
+
+const styles = {
+  /* Styles applied to the root element. */
+  root: {},
+  /* Styles applied to the ol element. */
+  ol: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    alignItems: 'center',
+    padding: 0, // Reset
+    margin: 0, // Reset
+    '& li': {
+      listStyle: 'none',
+    },
+  },
+  /* Styles applied to the separator element. */
+  separator: {},
+};
+
+class Breadcrumbs extends React.Component {
+  state = {
+    expanded: false,
+  };
+
+  handleClickExpand = () => {
+    this.setState({ expanded: true });
+  };
+
+  insertSeparators(items) {
+    return items.reduce((acc, current, index) => {
+      if (index < items.length - 1) {
+        acc = acc.concat(
+          current,
+          <BreadcrumbSeparator
+            // eslint-disable-next-line react/no-array-index-key
+            key={`separator-${index}`}
+            className={this.props.classes.separator}
+          >
+            {this.props.separator}
+          </BreadcrumbSeparator>,
+        );
+      } else {
+        acc.push(current);
+      }
+
+      return acc;
+    }, []);
+  }
+
+  renderItemsBeforeAndAfter(allItems) {
+    const { itemsBeforeCollapse, itemsAfterCollapse } = this.props;
+
+    // This defends against someone passing weird data, to ensure that if all
+    // items would be shown anyway, we just show all items without the EllipsisItem
+    if (itemsBeforeCollapse + itemsAfterCollapse >= allItems.length) {
+      return allItems;
+    }
+
+    return [
+      ...allItems.slice(0, itemsBeforeCollapse),
+      <BreadcrumbCollapsed key="ellipsis" onClick={this.handleClickExpand} />,
+      ...allItems.slice(allItems.length - itemsAfterCollapse, allItems.length),
+    ];
+  }
+
+  render() {
+    const {
+      children,
+      classes,
+      className: classNameProp,
+      component: Component,
+      itemsAfterCollapse,
+      itemsBeforeCollapse,
+      maxItems,
+      separator,
+      ...other
+    } = this.props;
+    const allItems = React.Children.toArray(children);
+
+    return (
+      <Component
+        aria-label="Breadcrumb navigation"
```

Comment:
I don't think we have a precedent for adding default labels. Either recommend that property in the docs (preferably link to https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html) or set it via `defaultProps` so that it's explicitly documented (and with some bikeshedding to use "Breadcrumb" since "navigation" is redundant on `nav` elements).

Reply:
I’d vote for moving this to the demos, and only setting aria-role.

Reply:
Yeah, ok. I'm happy with the "demo only" strategy.